GEOSwift/Codable/Point+Codable.swift

```diff
@@ -0,0 +1,15 @@
+// for internal use only; GeoJSON encoding & decoding helpers
+extension Point: CodableGeometry {
+    static let geoJSONType = GeoJSONType.point
+
+    var coordinates: [Double] {
+        return [x, y]
+    }
+
+    init(coordinates: [Double]) throws {
+        guard coordinates.count >= 2 else {
```

Comment:
I feel like maybe this should require exactly 2. I don't like that it silently ignores extra values, unless there's some case for passing a list of coordinates with other data where we only care about the first two values and that's obvious to the caller what they're doing.

Reply:
This is an interesting question.

The [GeoJSON spec section 3.1.1](https://tools.ietf.org/html/rfc7946#section-3.1.1) describes the semantics:

> A position is an array of numbers.  There MUST be two or more elements.  The first two elements are longitude and latitude, or easting and northing, precisely in that order and using decimal numbers.  Altitude or elevation MAY be included as an optional third element.

MAY is defined in RFC 2119:

> 5. MAY   This word, or the adjective "OPTIONAL", mean that an item is truly optional.  One vendor may choose to include the item because a particular marketplace requires it or because the vendor feels that it enhances the product while another vendor may omit the same item. An implementation which does not include a particular option MUST be prepared to interoperate with another implementation which does include the option, though perhaps with reduced functionality. In the same vein an implementation which does include a particular option MUST be prepared to interoperate with another implementation which does not include the option (except, of course, for the feature the option provides.)

This suggests to me that it's acceptable for us to ignore it, erring on the side of being lenient with what we accept.

OTOH, you bring up a good point that it could be surprising for the lib to silently drop the third value. Would it be more or less surprising for it to fail?

Options:

- Leave it as-is
- Throw if there are not exactly 2 values
- Make it configurable (I suppose we'd need to introduce some kind of settings object or global state and expose some way of configuring it).

Thoughts?

Reply:
In that case I prefer leaving this as-is so people don't have to explicitly grab only the first 2 values out of valid geojson data. If we find people are being surprised by the behavior of dropping altitude we can revisit then.

Reply:
@johnson-christopher if we did want to make this configurable, we could use a technique I learned about today. There's a `userInfo` dictionary on `JSONDecoder` that we could use to pass a flag/setting into the decoding process. Inside of our `Point.init(from: Deocder)` method, we could check the setting and use either the strict or lenient approach. We'd need to discuss which is a better default (I'd lean toward lenient).

Reply:
I do like that idea. I'd also lean toward making it lenient by default.

I'm not sure if it's worth putting into this release yet. I guess it depends on the effort. I'm fine with leaving it as-is for now, with this being on the roadmap for the future, or doing it now. Either way.

Reply:
Okay, let’s just defer it then. Easy to add later, especially if we don’t plan to change the default behavior.